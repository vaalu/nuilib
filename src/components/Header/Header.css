.defaultHeader {
	color: brown;
}
.blueHeader {
	color: blue;
}