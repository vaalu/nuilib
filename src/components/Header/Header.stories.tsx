import React from 'react'
import { storiesOf } from '@storybook/react'
import mocks from './__mocks__/Header.mock'
import { Header } from './Header'
storiesOf('Atom/Header', module)
	.add('H1', () => <Header {...mocks[0]}>{mocks[0].label}</Header>)
	.add('H2', () => <Header {...mocks[1]}>{mocks[1].label}</Header>)
	.add('H3', () => <Header {...mocks[2]}>{mocks[2].label}</Header>)
