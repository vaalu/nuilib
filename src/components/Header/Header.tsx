import React, { ReactNode } from 'react'
import './Header.css'
export type HeaderProps = {
	level:number
	children:ReactNode
	className?:string
}
export const Header = ({ level, className , children }:HeaderProps) => {
	const Heading = `h${ level }`
	const cssClass = { className : "defaultHeader"} 
	return React.createElement(Heading, cssClass, children)
}