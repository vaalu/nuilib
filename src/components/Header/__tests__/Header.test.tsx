import TestRenderer from 'react-test-renderer'
import React from 'react'
import { Header } from '../Header'
import mocks from '../__mocks__/Header.mock'
describe('Header', () => {
	it('Headers with levels test', () => {
		const tree = TestRenderer.create(
		<Header {...mocks[0]}>{ mocks[0].label }</Header>
		).toJSON()
		expect(tree).toMatchSnapshot()
	})
})