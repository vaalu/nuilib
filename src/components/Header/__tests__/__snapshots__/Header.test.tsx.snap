// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Headers with levels test 1`] = `
<h1
  className="defaultHeader"
>
  Header level 1
</h1>
`;
